pkg/rescaffold/migrate.go

```diff
@@ -285,18 +286,10 @@ func getAPIResourceFlags(resource resource.Resource) []string {
 		args = append(args, "--resource=false")
 	} else {
 		args = append(args, "--resource")
-		if resource.API.Namespaced {
-			args = append(args, "--namespaced")
-		} else {
-			args = append(args, "--namespaced=false")
-		}
+		args = append(args, fmt.Sprintf("--namespaced=%t", resource.API.Namespaced))
```

Comment:
Hi @fengshunli,

Thank you for your contribution 🥇

We appreciate your proposal to change the code.
However, we cannot move forward with this change. The Namespaced field in the specs is defined with the `omitempty` tag, see: 

https://github.com/fengshunli/kubebuilder/blame/136f6ce91fa090bcc988d40e37712bdcda8cd423/pkg/model/resource/api.go#L28-L29

This means that the field can be empty (i.e., it might not be present in the JSON representation if its value is false or not set). **As a result, the proposed change does not handle the case where resource.API.Namespaced is not present, which would be interpreted as false.**

Instead, we need to explicitly check for the presence and value of resource.API.Namespaced to ensure that the argument is correctly set. **As it is currently**

This ensures that the argument is set appropriately based on whether `resource.API.Namespaced` is true, false, or not present.